providers/plugin.rb

```diff
@@ -14,94 +14,66 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+use_inline_resources if defined?(:use_inline_resources)
```

Comment:
As a side note I'd love to convert this whole thing to a 12.5 custom resource, and then we can do away with a lot of this boilerplate.